cmd/bot/run.go

```diff
@@ -93,7 +93,7 @@ func ProcessUnprovenBotDelegatedWithdrawals(ctx context.Context, log log.Logger,
 	maxBlockTime := time.Now().Unix() - cfg.Misc.ProposeTimeWindow
 
 	unprovens := make([]core.L2ContractEvent, 0)
-	result := db.Order("id asc").Where("proven = false AND block_time < ? AND failure_reason IS NULL", maxBlockTime).Limit(limit).Find(&unprovens)
+	result := db.Order("id asc").Where("proven = false AND block_time < ? AND failure_reason = ''", maxBlockTime).Limit(limit).Find(&unprovens)
```

Comment:
Why do this change? Seems the schema still allows NULL.

Reply:
This is related to the Go type definition. The failure reason is defined as a Go string instead of a string. That makes `L2ContractEvent.FailureReason` default to be empty string `""` but not `nil`. Due to this, the failure reason inserted is `""` not `NULL`.

```go
type L2ContractEvent struct {
        ...
	FailureReason   string `gorm:"type:text"`
}
```

Reply:
I am gonna close this PR and fix this issue by changing `L2ContractEvent.FailureReason` to `*string`.